internal/storage/image.go

```diff
@@ -398,6 +401,43 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, nameOrID string) (bool, error) {
+	ref, err := svc.getRef(nameOrID)
+	if err != nil {
+		return false, fmt.Errorf("get ref for name or ID: %w", err)
+	}
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return false, fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return false, fmt.Errorf("create policy context: %w", err)
+	}
+
+	imageSource, err := ref.NewImageSource(ctx, systemContext)
+	if err != nil {
+		return false, fmt.Errorf("new image source: %w", err)
+	}
+	defer imageSource.Close()
+
+	storeImage, err := istorage.Transport.GetStoreImage(svc.store, ref)
+	if err != nil {
+		return false, fmt.Errorf("get store image: %w", err)
+	}
+
+	unparsedImage := cimage.UnparsedInstance(imageSource, &storeImage.Digest)
+
+	isAllowed, err := policyContext.IsRunningImageAllowed(ctx, unparsedImage)
```

Comment:
This is using a policy for `containers-storage:`, not for the relevant registry. And that policy typically allows everything (or, if not configured so, falls back to the generic global default.)

---

In general, verifying the signature on an already-locally-pulled-image, is… interesting. We do store copies of the signatures locally, so it should be _possible_, but
- right now that would require a ~hack, creating a custom `ImageReference` implementation that points to a registry but fetches data locally. Ugly but viable
- we do not _always_ manage to preserve the originally-signed manifest: https://github.com/containers/image/issues/1768 , and encryption is a supported CRI-O feature. So those images would just break.

OTOH it _could_ be quite a bit simpler, and thus, at least in a sense, less risky, than contacting the registry and then trying to match the two images (and dealing with tags that move concurrently with our pull+start and the like).